src/main/java/frc/robot/Auto.java

```diff
@@ -114,10 +127,84 @@ public void runDefaultAutonomous() {
         }
     }
 
+    public void runBlueACubeLeaveCommunity() {
+        if (timer.get() < Blue_ACubeLeaveCommunity.getTotalTimeSeconds()) {
+            State desiredPose = Blue_ACubeLeaveCommunity.sample(timer.get());
+            ChassisSpeeds refChassisSpeeds = m_ramseteController.calculate(drivetrain.getPose(), desiredPose);
+
+            drivetrain.autoDrive(refChassisSpeeds.vxMetersPerSecond, refChassisSpeeds.omegaRadiansPerSecond);
+            System.out.println(Blue_ACubeLeaveCommunity.getTotalTimeSeconds()-timer.get());
+
+        } else {
+            drivetrain.autoDrive(0, 0);
+            System.out.println("FINISHED");
+        }
+    }
+
+    public void runBlueBlueACubeBalance() {
+        if (timer.get() < Red_ACubeBalance.getTotalTimeSeconds()) {
+            State desiredPose = Red_ACubeBalance.sample(timer.get());
+            ChassisSpeeds refChassisSpeeds = m_ramseteController.calculate(drivetrain.getPose(), desiredPose);
+
+            drivetrain.autoDrive(refChassisSpeeds.vxMetersPerSecond, refChassisSpeeds.omegaRadiansPerSecond);
+            System.out.println(Red_ACubeBalance.getTotalTimeSeconds()-timer.get());
+
+        } else {
+            drivetrain.autoDrive(0, 0);
+            System.out.println("FINISHED");
+            finishedTrajectory = true;
+        }
+    }
+
+    public void runBlueBCubeBalance() {
+        if (timer.get() < Blue_BCubeBalance.getTotalTimeSeconds()) {
+            State desiredPose = Blue_BCubeBalance.sample(timer.get());
+            ChassisSpeeds refChassisSpeeds = m_ramseteController.calculate(drivetrain.getPose(), desiredPose);
+
+            drivetrain.autoDrive(refChassisSpeeds.vxMetersPerSecond, refChassisSpeeds.omegaRadiansPerSecond);
+            System.out.println(Blue_BCubeBalance.getTotalTimeSeconds()-timer.get());
+
+        } else {
+            drivetrain.autoDrive(0, 0);
+            System.out.println("FINISHED");
+        }
+    }
+
+    public void runBlueCCubeLeaveCommunity() {
+        if (timer.get() < Blue_CCubeLeaveCommunity.getTotalTimeSeconds()) {
+            State desiredPose = Blue_CCubeLeaveCommunity.sample(timer.get());
+            ChassisSpeeds refChassisSpeeds = m_ramseteController.calculate(drivetrain.getPose(), desiredPose);
+
+            drivetrain.autoDrive(refChassisSpeeds.vxMetersPerSecond, refChassisSpeeds.omegaRadiansPerSecond);
+            System.out.println(Blue_CCubeLeaveCommunity.getTotalTimeSeconds()-timer.get());
+
+        } else {
+            drivetrain.autoDrive(0, 0);
+            System.out.println("FINISHED");
+        }
+    }
+
+    public void runBlueCCubeBalance() {
```

Comment:
if you passed the trajectory in as a parameter, these five methods could be one:  runTrajectory(PathPlannerTrajectory trajectory) {...}